evaluator/src/main/kotlin/Evaluator.kt

```diff
@@ -40,12 +40,21 @@ class Evaluator(
     licenseInfoResolver: LicenseInfoResolver = ortResult.createLicenseInfoResolver(),
     resolutionProvider: ResolutionProvider = DefaultResolutionProvider.create(),
     licenseClassifications: LicenseClassifications = LicenseClassifications(),
+    projectSourceTree: SourceTreeResolver = SourceTreeResolver.forRemoteRepository(ortResult.repository.vcsProcessed),
```

Comment:
So, the whole script evaluation still gets a `SourceTreeResolver`... however, I was hoping to see a solution that does not change any host-side scripting classes at all, but only adds a new rule class derived from `OrtResultRule` which creates a `SourceTreeResolver` from the `OrtResult`.

Am I overlooking something why that's not possible? Do you want me to make a proposal based on your code?

Reply:
> Am I overlooking something why that's not possible? Do you want me to make a proposal based on your code?



This code is using constructor injection of dependencies for testability. E.g. the test can inject an arbitrary directory as source code directory. I understand that you dislike constructor injection in this case. So, can you tell which way of injecting a directory from the test would be acceptable for you? (Ideally along with an argumentation why)

Having the above question answered would unblock the situation. Note that I understood from previous talks that you didn't like the construct, however I didn't get why and also not what alternative you want - which is why I didn't change it. It just didn't make sense to me.

Reply:
I changed things so that it fits what I understand you have in mind. I believe all should be covered now, except for the following two things for which I was lacking your input, so I did my best guess as follows:

1. I used injection via setter override, as to my understanding you seem to dislike injecting the resolver into the constructor or the corresponding factory method. Please see if you like that more.
2. I used the name `OrtResultRule2`. Please propose a better name as you pushed for introducing this class.

Reply:
> So, can you tell which way of injecting a directory from the test would be acceptable for you? (Ideally along with an argumentation why)

If there was a rule that inherits from `OrtResultRule` and that takes a `projectSourceTree` as a constructor argument, wouldn't that work to run the tests on that new rule, passing in "fake" sources to run the test on?

> Note that I understood from previous talks that you didn't like the construct, however I didn't get why

My main arguments for not passing a `projectSourceTree` to the `Evaluator`'s constructor are:
- Not every rule needs this, and evaluating it / cloning the project source code (accidentally) is expensive.
- The information for a `SourceTree` can be derived from the `ortResult` which already gets passed. IMO there is no need (and it's bad design) to extract some more or less arbitrary bit of information from the `ortResult` and pass a class building upon it independently.

Reply:
> Passing a `SourceTree` to the constructor of `OrtResultRule2` would be of course fine! Doesn't that work?

Hm, I see. In the previous versions of my change it was a problem that this `SourceTreeResolver` was part of the user facing API. Did I just misunderstand that? (No worries, I'll change the code now to pass it into the constructor)

> I believe the proposal I had mentioned in our call earlier this week was ProjectSourceRule.

Let me just rename it quickly.